sentry/sentry.go

```diff
@@ -0,0 +1,83 @@
+package sentry
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/fabric8-services/fabric8-wit/token"
+
+	"github.com/getsentry/raven-go"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+)
+
+// Client encapsulates client to Sentry service
+// also has mutex which controls access to the client
+type Client struct {
+	c   *raven.Client
+	mux sync.Mutex
+}
+
+var (
+	sentryClient *Client
+)
+
+// Sentry returns client declared inside package
+func Sentry() *Client {
+	return sentryClient
+}
+
+// InitializeSentryClient initializes sentry client
+func InitializeSentryClient(release string) error {
+	c, err := raven.New(os.Getenv("SENTRY_DSN"))
+	if err != nil {
+		return err
+	}
+
+	c.SetRelease(release)
```

Comment:
We would need Environment in here as well. Not sure if we have any good definition of that already, or if we need to expose a new config option. @alexeykazakov thoughts?

Reply:
Yeah.. We could look up the domain in the request to map - "openshift.io" - "prod", "prod-preview.openshift.io" - "preview/staging" but we don't have the incoming request in main(). So, we have to create a new config option. But we could just rely on AuthURL by default, so, we don't have to create a new config map for this env. Something like:

configuration.go
```go
const varEnvironment = "environment"
...
func (c *ConfigurationData) GetEnvironment() string {
        env := c.v.GetString(varEnvironment)
        if env!="" {
               return env
        }
        authURL := c.GetAuthURL()
        // Parse the URL and if it's openshift.io then return "prod"
        // if it's prod-preview.openshift.io then return preview.
        // Otherwise return "local" or something.
	...
}
```
Latter we could explicitly set the env via F8_ENVIRONMENT in the OS template and CM if needed.

Reply:
done